modules/xfem/src/efa/EFAEdge.C

```diff
@@ -0,0 +1,419 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "EFAEdge.h"
+
+#include "EFANode.h"
+#include "EFAError.h"
+
+EFAEdge::EFAEdge(EFANode * node1, EFANode * node2):
+  _edge_node1(node1),
+  _edge_node2(node2)
+{
+  _embedded_nodes.clear();
+  _intersection_x.clear();
+  consistencyCheck();
+}
+
+EFAEdge::EFAEdge(const EFAEdge & other_edge)
+{
+  _edge_node1 = other_edge._edge_node1;
+  _edge_node2 = other_edge._edge_node2;
+  _intersection_x = other_edge._intersection_x;
+  _embedded_nodes = other_edge._embedded_nodes;
+  consistencyCheck();
+}
+
+EFAEdge::~EFAEdge() // do not delete edge node - they will be deleted
+{}                  // in EFAelement's destructor
```

Comment:
Yeah, I'm not sure I like the sound of that ;-). Smart pointers are most likely the better way here.